packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts

```diff
@@ -65,23 +65,49 @@ export function getSPOAndGraphRequestIdsFromResponse(headers: { get: (id: string
 
 export interface IFacetCodes {
     facetCodes?: string[];
- }
+}
 
-export function parseFacetCodes(response: string): string[] {
-    const stack: string[] = [];
-    let error;
-    try {
-        error = JSON.parse(response).error;
+/** Empirically-based model of error response inner error from ODSP */
+interface OdspErrorResponseInnerError {
+    code: string;
+    innerError?: OdspErrorResponseInnerError
+}
+
+/** Empirically-based model of error responses from ODSP */
+interface OdspErrorResponse {
+    error: OdspErrorResponseInnerError & {
+        message: string;
     }
-    catch(e) {
-        return stack;
+}
+
+/** Empirically-based type guard for error responses from ODSP */
+function isOdspErrorResponse(x: any): x is OdspErrorResponse {
+    const error = x?.error;
+    return typeof(error?.message) === "string" &&
+        typeof(error?.code) === "string";
+}
+
+function tryParseErrorResponse(
+    response: string | undefined,
+): { success: true, errorResponse: OdspErrorResponse } | { success: false } {
+    try {
+        if (response !== undefined) {
+            const parsed = JSON.parse(response);
+            if (isOdspErrorResponse(parsed)) {
+                return { success: true, errorResponse: parsed };
+            }
+        }
     }
+    catch(e) {}
+    return { success: false };
+}
 
+function parseFacetCodes(errorResponse: OdspErrorResponse): string[] {
+    const stack: string[] = [];
+    let error: OdspErrorResponseInnerError | undefined = errorResponse.error;
     // eslint-disable-next-line no-null/no-null
     while (typeof error === "object" && error !== null) {
-        if (error.code !== undefined) {
```

Comment:
I checked and in 100% of the cases in the logs (30 day retention or so), if `message` is set on the response error then `code` is as well.  Hence the type guard above checking for both, and we can confidently remove this additional check.

Reply:
Reverted it back after seeing UTs handle it, it's no big deal I think.